e2e-tests/test/hardhat-apis.test.ts

```diff
@@ -0,0 +1,104 @@
+import { expect } from "chai";
+import { Wallet } from "zksync-web3";
+import { getTestProvider } from "../helpers/utils";
+import { RichAccounts } from "../helpers/constants";
+import { ethers } from "hardhat";
+
+const provider = getTestProvider();
+
+describe("hardhat_setBalance", function () {
+  it("Should update the balance of an account", async function () {
+    // Arrange
+    const userWallet = Wallet.createRandom().connect(provider);
+    const newBalance = ethers.utils.parseEther("42");
+
+    // Act
+    await provider.send("hardhat_setBalance", [
+      userWallet.address,
+      newBalance._hex,
+    ]);
+
+    // Assert
+    const balance = await userWallet.getBalance();
+    expect(balance.eq(newBalance)).to.true;
+  });
+});
+
+describe("hardhat_setNonce", function () {
+  it("Should update the nonce of an account", async function () {
+    // Arrange
+    const userWallet = Wallet.createRandom().connect(provider);
+    const newNonce = 42;
+
+    // Act
+    await provider.send("hardhat_setNonce", [
+      userWallet.address,
+      ethers.utils.hexlify(newNonce),
+    ]);
+
+    // Assert
+    const nonce = await userWallet.getNonce();
+    expect(nonce).to.equal(newNonce);
+  });
+});
+
+describe("hardhat_mine", function () {
+  it("Should mine multiple blocks with a given interval", async function () {
+    // Arrange
+    const numberOfBlocks = 100;
+    const intervalInSeconds = 60;
+    const startingBlock = await provider.getBlock("latest");
+    const startingTimestamp: number = await provider.send(
+      "config_getCurrentTimestamp",
+      []
+    );
+
+    // Act
+    await provider.send("hardhat_mine", [
+      ethers.utils.hexlify(numberOfBlocks),
+      ethers.utils.hexlify(intervalInSeconds),
+    ]);
+
+    // Assert
+    const latestBlock = await provider.getBlock("latest");
+    expect(latestBlock.number).to.equal(
+      startingBlock.number + numberOfBlocks,
+      "Block number mismatch"
+    );
+    expect(latestBlock.timestamp).to.equal(
+      startingTimestamp + (numberOfBlocks - 1) * intervalInSeconds * 1000,
+      "Timestamp mismatch"
+    );
+  });
+});
+
+// TODO: Run this test once eth_sendTransaction has been implemented or when different tests have been added
```

Comment:
maybe comment out the whole test body? (so that it is clear that this test is disabled)

Reply:
This test shows as disabled in the output of the test run when `xdescribe` is added